### PR TITLE
Fix that callback of Post.render () is not used

### PR DESCRIPTION
**Pull request by segayuu**
I got callback as an argument, but ignored it.
Along with this correction, refactoring is carried out.

- [x] Add test cases for the changes.
- [ ] Passed the CI test.

### CODE REVIEW

**Comment by NoahDragon** [comment]
Is this still working in progress?
Could you please elaborate the reason behind this change?

**Comment by segayuu** [comment]
thank @NoahDragon .
I decided to implement because callback was unnecessary argument or it was not possible to determine whether I forgot to implement it.

File: lib/hexo/post.js [hunk 1]
```diff
@@ -225,11 +197,14 @@ Post.prototype.publish = function(data, replace, callback) {
   }).thenReturn(result).asCallback(callback);
 };
 
-Post.prototype.render = function(source, data = {}, callback) {
+Post.prototype.render = function(source, data = {}, callback = null) {
+  if (callback !== null && typeof callback !== 'function') {
+    throw new TypeError('callback most function');
```
Comment (JLHwung) [inline lib/hexo/post.js]: Should it be `callback must be a function`?

**Comment by JLHwung** [comment]
Great job! If you don't mind it would be even better if you can split the refactor code and bugfix/feature codes into separate commits, thus reviewers can be more focused alongside the organized commit history.

**Comment by segayuu** [comment]
@JLHwung 
When doing it with this PR, I am afraid to do it because `git push --force` is needed.
I feel comfortable to rebuild PR :)

**Comment by JLHwung** [comment]
@segayuu You can if you are only force pushing your own development branch. `git push --force` only bothers in a collaborative situation.

Regards to this PR, you can leave it be and take in mind for the following PRs. :-)

**Comment by segayuu** [comment]
@JLHwung I did `git push --force` and completed organizing the commit history.
It is appreciated if you can review again.

**Comment by JLHwung** [comment]
Awesome!

**Comment by NoahDragon** [comment]
@segayuu Just a quick question, will this PR fix the https://github.com/hexojs/hexo/issues/2451 issue? The behaviors(logic) between `hexo g` and `hexo s` are different.

**Comment by segayuu** [comment]
I am not so cognizant that I can debug with multiple classes. (So I am only changing it with a simple substitution, or changing it to fit within one file.)
I do not think there will be a day when I can clearly understand whether it is a problem of hexo-cli or hexo (core) · · ·.

**Pull request merged**